sdk/terraform/Azure.ResourceManager.Terraform/api/Azure.ResourceManager.Terraform.netstandard2.0.cs

```diff
@@ -0,0 +1,146 @@
+    }
+}
+namespace Azure.ResourceManager.Terraform.Mocking
+{
+    public partial class MockableTerraformSubscriptionResource : Azure.ResourceManager.ArmResource
+    {
+        protected MockableTerraformSubscriptionResource() { }
+        public virtual Azure.ResourceManager.ArmOperation<Azure.ResourceManager.Terraform.Models.TerraformOperationStatus> ExportTerraform(Azure.WaitUntil waitUntil, Azure.ResourceManager.Terraform.Models.BaseExportModel body, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.Terraform.Models.TerraformOperationStatus>> ExportTerraformAsync(Azure.WaitUntil waitUntil, Azure.ResourceManager.Terraform.Models.BaseExportModel body, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+    }
+}
+namespace Azure.ResourceManager.Terraform.Models
+{
+    public static partial class ArmTerraformModelFactory
+    {
+        public static Azure.ResourceManager.Terraform.Models.ExportQuery ExportQuery(Azure.ResourceManager.Terraform.Models.TargetProvider? targetProvider = default(Azure.ResourceManager.Terraform.Models.TargetProvider?), bool? fullProperties = default(bool?), bool? maskSensitive = default(bool?), string query = null, string namePattern = null, bool? recursive = default(bool?)) { throw null; }
+        public static Azure.ResourceManager.Terraform.Models.ExportResourceGroup ExportResourceGroup(Azure.ResourceManager.Terraform.Models.TargetProvider? targetProvider = default(Azure.ResourceManager.Terraform.Models.TargetProvider?), bool? fullProperties = default(bool?), bool? maskSensitive = default(bool?), string resourceGroupName = null, string namePattern = null) { throw null; }
+        public static Azure.ResourceManager.Terraform.Models.ExportResult ExportResult(string configuration = null, System.Collections.Generic.IEnumerable<string> skippedResources = null, System.Collections.Generic.IEnumerable<Azure.ResponseError> errors = null) { throw null; }
+        public static Azure.ResourceManager.Terraform.Models.TerraformOperationStatus TerraformOperationStatus(Azure.ResourceManager.Terraform.Models.ExportResult properties = null, Azure.ResourceManager.Terraform.Models.ResourceProvisioningState status = default(Azure.ResourceManager.Terraform.Models.ResourceProvisioningState), string name = null, System.DateTimeOffset? startOn = default(System.DateTimeOffset?), System.DateTimeOffset? endOn = default(System.DateTimeOffset?), double? percentComplete = default(double?), Azure.ResponseError error = null) { throw null; }
+    }
+    public abstract partial class BaseExportModel : System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.Terraform.Models.BaseExportModel>, System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Terraform.Models.BaseExportModel>
```

Comment:
usually in .Net SDK we would like to avoid the word `base`.
Could this be `TerraformExportContent`?

Reply:
`BaseExportModel` is inteded to hold common properties of all kind of Azure -> Terraform exports: `ExportQuery`, `ExportResource` and `ExportResourceGroup`.

Refer to the CLI help:

```
$  az  azure-terraform  export-terraform --help
...
ExportParameter Arguments
    --export-query          : Export parameter for resources queried by ARG (Azure Resource Graph)
                              Support shorthand-syntax, json-file and yaml-file. Try "??" to show
                              more.
    --export-resource       : Export parameter for single resources  Support shorthand-syntax, json-
                              file and yaml-file. Try "??" to show more.
    --export-resource-group : Export parameter for a resource group  Support shorthand-syntax, json-
                              file and yaml-file. Try "??" to show more.
```

Would `CommonExportProperties` work @ArcturusZhang ?

Reply:
Renamed to `CommonExportProperties` in https://github.com/Azure/azure-rest-api-specs/pull/31198